File: src/main/java/MusicCube/entities/Artist.java
package MusicCube.entities;


import javax.persistence.Column;
import javax.persistence.Entity;
import java.util.Date;

@Entity
public class Artist extends Person {

    @Column
    private String stageName;

    public Artist() {}

    public Artist(Localization origin, String firstNames, String lastName, Date birthDate, String stageName) {
        super(origin, firstNames, lastName, birthDate);
        this.stageName = stageName;
    }

    public String getStageName() {
        return stageName;
    }

    public void setStageName(String stageName) {
        this.stageName = stageName;
    }
}

File: src/app/Class/Artist.ts
import { Person } from "./Person";

export class Artist extends Person {
  private stageName: string;

  constructor(obj?: any) {
    super(obj);
    this.stageName = (obj && obj.stageName) || "";
  }

  // Getter

  getStageName(): string {
    return this.stageName;
  }

  // Setter

  setStageName(stageName: string) {
    this.stageName = stageName;
  }
}
